### DISCUSSION TITLE
Building New UI using react with same sunbird Ed backend

### DISCUSSION THREAD

**Discussion by paritshivani** [Sunbird-Ed/Community#679] [Sunbird ED configuration discussion]
Hi team, 
Hope you are doing well!
Here we have came up with new approach of using sunbird ED. Where we are trying to build new UI for existing sunbird Ed portal features using react! this will be lighter version of current system.
For this we have few questions we want help on:

- We are creating own login page here with react. As per my understading, We are not using jwt token given by keyclock directly in sunbird and we have some customisations there. what exaclty we have to do instead of using jwt token given by login api response of keyclock.

- Until our React UI is fully mapped with current Angular UI. We need to give access of both UIs' to users. So my question is , how we can share user session in two sub-domain if we give UI switch to users on screen and impliment seemless transition between old Angular UI and new react UI

@vinukumar-vs ,@rajeevsathish, @AmiableAnil ,  @Krishnaj20, @chitranshu-keshav, @SanthoshVasabhaktula, gandham-santhosh, @coolbung @komalm

**Comment by ManojNathIC** [comment]
Hi Team,

I'm searching for a Sunbird user login API that returns a JWT token for authenticated users. 
Specifically, I'm interested in information regarding the Sunbird Keycloak login API or any similar APIs for user authentication.

I appreciate any insights. Thank you!

**Reply by vinukumar-vs** [reply]
@ManojNathIC Please refer the below Sunbird.Lern micorsite API documentation 
https://lern.sunbird.org/use/developer-guide/user-and-org-service/apis/user-management#user-v3-create

Please refer https://bot.sunbird.org/ response. You can ask more relevant questions, it will help to get related response
<img width="1200" alt="image" src="https://github.com/Sunbird-Ed/Community/assets/4496096/e273a723-af02-4d10-8e6b-325e381761ea">

I would suggest to use Sunbird-Lern BB discussions for login related queries
[@AmiableAnil](https://github.com/AmiableAnil) @mohitga

**Reply by vinukumar-vs** [reply]
As discussed yesterday, there are aproches to it
1. add the React application build into the same `dist` folder along with exiting portal build. Load this react build using different route for react application handed by the backend portal
2. Second approach is by using NPM package(web component of react application) loading in the front-end portal, similar to other libraries like UCI-admin, Discussion-UI etc.

As agreed, do a quick POC with the first approach.

@mohitga @coolbung @komalm
